zerver/tests/test_auth_backends.py

```diff
@@ -404,419 +405,464 @@ def json(self) -> str:
     def text(self) -> str:
         return "Response text"
 
-class GitHubAuthBackendTest(ZulipTestCase):
-    def setUp(self) -> None:
-        self.user_profile = self.example_user('hamlet')
-        self.email = self.user_profile.email
-        self.name = 'Hamlet'
-        self.backend = GitHubAuthBackend()
-        self.backend.strategy = DjangoStrategy(storage=BaseDjangoStorage())
-        self.user_profile.backend = self.backend
+class SocialAuthBaseWrapper:
+    # This is a hack so that tests in this base class are not run.
+    class SocialAuthBase(ZulipTestCase):
+        """This is a base class for testing social-auth backends. Following
+        methods can be overrided by subclasses as per the backend:
+
+            setUpHook() - If any subclass wants to do some initialization
+                        during `setUp()` phase then they may add their
+                        logic here.
```

Comment:
Maybe worth mentioning that we use this rather than subclasses calling super because of the workaround in `setUp`